ecl/hql/hqlgram2.cpp

```diff
@@ -11958,16 +11985,15 @@ IHqlExpression *HqlGram::yyParse(bool _parsingTemplateAttribute, bool catchAbort
     }
     catch (IException * e)
     {
-        if (e->errorCode() != ERR_ABORT_PARSING)
+        if (catchAbort || e->errorCode() != ERR_ABORT_PARSING)
```

Comment:
It looks much better if this error is not reported.  E.g., see the following sample:

err2092.ecl(24,13): error C3002: syntax error near "aaa" : expected RANGE, ROWSET, SELF, SUCCESS, datarow, dataset, dictionary, module-name, identifier, identifier, function-name, identifier, macro-name, '^', '(', '['
err2092.ecl(24,16): error C3001: Too many errors (max = 5)
err2092.ecl(0,0): error C2000: Parsing aborted

The parsing aborted line does not add anything useful.

The previous text is a bit clearer

Too many errors; parsing aborted

You could change it to

Too many errors (n); parsing aborted